torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -712,29 +712,32 @@ FusionExecutor::GlobalBuffers FusionExecutor::allocGlobalVals(
 }
 
 std::vector<at::Tensor> FusionExecutor::allocOutputs(
+    const KernelArgumentHolder& args,
     kir::ExpressionEvaluator& expr_eval,
     const std::unordered_set<int>& alias_indices) {
   FUSER_PERF_SCOPE("FusionExecutor::AllocOutputs");
   const auto kernel = lowered_->kernel();
   // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
   std::vector<at::Tensor> outputs;
   for (const auto out_i : c10::irange(kernel->outputs().size())) {
-    // TODO: FIX this short-cut where we trivially forward inputs to outputs
     if (kernel->outputs()[out_i]->isFusionInput()) {
-      TORCH_INTERNAL_ASSERT(false, "trivial input forwarding NOT IMPLEMENTED");
-      // for (auto inp_i : c10::irange(kernel->inputs().size())) {
-      //   if (kernel->inputs()[inp_i] == kernel->outputs()[out_i]) {
-      //     TORCH_INTERNAL_ASSERT(
-      //         inp_i < inputs.size(),
-      //         "Issue with an input showing up as output, couldn't find
-      //         input.");
-      //     TORCH_INTERNAL_ASSERT(
-      //         inputs[inp_i].isTensor(),
-      //         "Cannot register a scalar as an output in a fusion.");
-      //     outputs.push_back(inputs[inp_i].toTensor());
-      //     break;
-      //   }
-      // }
+      for (auto inp_i : c10::irange(kernel->inputs().size())) {
+        if (kernel->inputs()[inp_i] == kernel->outputs()[out_i]) {
+          TORCH_INTERNAL_ASSERT(
+              inp_i < args.size(),
+              "Issue with an input showing up as output, couldn't find input.");
+          TORCH_INTERNAL_ASSERT(
+              args[inp_i]->isType(ArgType::Tensor),
+              "Cannot register a scalar as an output in a fusion.");
+          // pushing empty tensor for trivial forwarding. Since we handle this
+          // in integration, see step 1 - note [trivial forwarding]
+          c10::Device device(c10::DeviceType::CUDA, args.getDeviceIndex());
+          const auto tensor_options =
+              at::TensorOptions().dtype(at::kFloat).device(device);
+          outputs.emplace_back(at::empty({0}, tensor_options));
+          break;
+        }
+      }
     } else {
```

Comment:
This is slightly different. We get away with an `emplace_back` because we'll swap the aliased input here before passing arguments to validation checks.
Of course it is still functional if we pass an empty tensor as with the case above. So we can certainly merge the two. I feel that messes up the logic and make the code hard to read?!

Feel free to ping this again if you have strong feeling on merging the two. I can do that and just put more comment inline.

Reply:
I don't understand what is the difference between `alias` vs `trivial forwarding`? Am I correct that, if an output is an alias of another output, then it is an `alias`, and if it is an alias of an input, the it is `trivial forwarding`?

Reply:
Discussed it offline. marking this one as resolved.